container-chains/nodes/simple/src/command.rs

```diff
@@ -592,3 +499,152 @@ impl CliConfiguration<Self> for RelayChainCli {
+                .map(|e| e.para_id)
+                .ok_or("Could not find parachain ID in chain-spec.")?;
+
+            let para_id = ParaId::from(para_id);
+
+            // TODO: Once there is an embeded node this should use it.
+            let keystore_container = KeystoreContainer::new(&config.keystore)?;
+
+            let collator_options = cli.run.collator_options();
+
+            let polkadot_cli = RelayChainCli::new(
+                &config,
+                [RelayChainCli::executable_name()]
+                    .iter()
+                    .chain(cli.relaychain_args().iter()),
+            );
+
+            let tokio_handle = config.tokio_handle.clone();
+            let polkadot_config =
+                SubstrateCli::create_configuration(&polkadot_cli, &polkadot_cli, tokio_handle)
+                    .map_err(|err| format!("Relay chain argument error: {}", err))?;
+
+            let telemetry = config
+                .telemetry_endpoints
+                .clone()
+                .filter(|x| !x.is_empty())
+                .map(|endpoints| -> std::result::Result<_, sc_telemetry::Error> {
+                    let worker = TelemetryWorker::new(16)?;
+                    let telemetry = worker.handle().new_telemetry(endpoints);
+                    Ok((worker, telemetry))
+                })
+                .transpose()
+                .map_err(sc_service::Error::Telemetry)?;
+
+            let telemetry_worker_handle = telemetry.as_ref().map(|(worker, _)| worker.handle());
+
+            let (relay_chain_interface, _collation_pair) = build_relay_chain_interface(
+                polkadot_config,
+                &config,
+                telemetry_worker_handle,
+                &mut task_manager,
+                collator_options,
+                None,
+            )
+            .await
+            .map_err(|e| sc_service::Error::Application(Box::new(e) as Box<_>))?;
+
+            let relay_chain = crate::chain_spec::Extensions::try_get(&*config.chain_spec)
+                .map(|e| e.relay_chain.clone())
+                .ok_or("Could not find relay_chain extension in chain-spec.")?;
+
+            let container_chain_spawner = ContainerChainSpawner {
+                params: ContainerChainSpawnParams {
+                    orchestrator_chain_interface,
+                    container_chain_cli,
+                    tokio_handle: config.tokio_handle.clone(),
+                    chain_type: config.chain_spec.chain_type(),
+                    relay_chain,
+                    relay_chain_interface,
+                    sync_keystore: keystore_container.keystore(),
+                    orchestrator_para_id: para_id,
+                    collation_params: None,
+                    spawn_handle: task_manager.spawn_handle().clone(),
+                    // TODO: Which sync mode to use?
+                    sync_mode: { move |_db_exists, _para_id| Ok(sc_cli::SyncMode::Warp) },
```

Comment:
You can reuse the same logic as the other function, right?

```rust
move |db_exists, para_id| {
    spawner::select_sync_mode_using_client(
        db_exists,
        &orchestrator_client.clone(),
        para_id,
    )
}
```

Reply:
No because it depends on the client which we don't have.

Reply:
Since this is not a collator it can safely use warp sync.

Because the bug is collators getting stuck trying to warp sync a chain with 0 blocks and not producing blocks. But if the data preserver node gets stuck if will fix itself when another node creates a new block.

Reply:
*disregard*